### ISSUE TITLE
GridPro'sd custom()-method should be overloaded with lambda for component

### ISSUE THREAD

**Issue by finaris-cs** [vaadin/vaadin-grid-pro#150]
Currently, it is not possible to use a component based on the item in the current row.

From API-point of view, it would be nice if a lamba expression could be passed that defines the component to render

**Comment by stefanuebe** [comment]
An example, what we would need in a project (I assume that is what the author means): 

```
grid.addEditColumn(Sample::getDate)
    .custom(item -> {
            if (item.isTimeNeeded()) {
            return new DateTimePicker();
        }
        return new DatePicker();
}, Sample::setDate);
```

**Comment by finaris-cs** [comment]
yes, what stefanuebe posted is exactly what i need. Currently, the custom() - Method only allows an AbstractField to be passed, but not a lambda / java.util.Function<GRIDITEMTYPE, AbstractField>
